tests/adapter/test_dataclass.py

```diff
@@ -674,3 +674,146 @@ def test_list():
             }
         ),
     )
+
+
+def test_dataclass_repr():
+
+    Example(
+        """\
+from inline_snapshot import snapshot,Is
+from dataclasses import dataclass,field
+
+@dataclass
+class container:
+    a: int
+
+    def __repr__(self):
+        return f"Repr {self.a}"
+
+def test_list():
+    l=container(5)
+    assert l == snapshot(l), "not equal"
+"""
+    ).run_inline(
+        ["--inline-snapshot=update"],
+        changed_files=snapshot(
+            {
+                "test_something.py": """\
+from inline_snapshot import snapshot,Is
+from dataclasses import dataclass,field
+
+@dataclass
+class container:
+    a: int
+
+    def __repr__(self):
+        return f"Repr {self.a}"
+
+def test_list():
+    l=container(5)
+    assert l == snapshot(container("Repr 5")), "not equal"
+"""
+            }
+        ),
+    )
+
+
+def test_dataclass_customize_repr():
+
+    Example(
+        """\
+from inline_snapshot import snapshot,Is,customize_repr
+from dataclasses import dataclass,field
+
+@dataclass
+class container:
+    a: int
+
+    def __repr__(self):
+        return f"Repr {self.a}"
+
+@customize_repr
+def _(value:container):
+    return f"Custom Repr {value.a}"
+
+def test_list():
+    l=container(5)
+    assert l == snapshot(l), "not equal"
+"""
+    ).run_inline(
+        ["--inline-snapshot=update"],
+        changed_files=snapshot(
+            {
+                "test_something.py": """\
+from inline_snapshot import snapshot,Is,customize_repr
+from dataclasses import dataclass,field
+
+
+@dataclass
+class container:
+    a: int
+
+    def __repr__(self):
+        return f"Repr {self.a}"
+
+@customize_repr
+def _(value:container):
+    return f"Custom Repr {value.a}"
+
+def test_list():
+    l=container(5)
+    assert l == snapshot(container("Custom Repr 5")), "not equal"
+"""
+            }
+        ),
+    )
+
+
+def test_dataclass_repr_for_func():
+
+    Example(
+        """\
+from inline_snapshot import snapshot,Is
+from dataclasses import dataclass,field
+from collections.abc import Callable
+
+def func():...
+
+@dataclass
+class container:
+    a: int
+    b: Callable
+
+    def __repr__(self):
+        return f"Repr {self.a} {self.b.__name__}"
+
+def test_list():
+    l=container(5, func)
+    assert l == snapshot(l), "not equal"
+"""
+    ).run_inline(
+        ["--inline-snapshot=update"],
+        changed_files=snapshot(
+            {
+                "test_something.py": """\
+from inline_snapshot import snapshot,Is
+from dataclasses import dataclass,field
+from collections.abc import Callable
+
+def func():...
+
+@dataclass
+class container:
+    a: int
+    b: Callable
+
+    def __repr__(self):
+        return f"Repr {self.a} {self.b.__name__}"
+
+def test_list():
+    l=container(5, func)
+    assert l == snapshot(container("Repr 5")), "not equal"
```

Comment:
produces `assert l == snapshot(container(a=5, b=HasRepr(function, "<function func at 0x10a8b4180>"))), "not equal"`

Reply:
maybe something like this:
``` python
from types import FunctionType
from inline_snapshot import snapshot, customize_repr


@customize_repr
def _(f:FunctionType):
    return f.__qualname__


def f():pass

class A:
    def g(self):pass


def test_a():
    assert f==snapshot(f)
    assert A.g==snapshot(A.g)
```

Reply:
Great!
I've slightly changed the function customize to be:
```
@customize_repr
def _(f: FunctionType):
    if f.__module__:
        return f"{f.__module__}.{f.__qualname__}"
    return f"{f.__qualname__}"
```

Is it something that should be standard for FunctionTypes?